src/FilterService/ListHelper.php

```diff
@@ -67,18 +68,22 @@ public function setupProductList(
             $params['currentPage'] = (int)$params['page'];
             $offset = $pageLimit * ($params['page'] - 1);
         }
+
+        $productList->setLimit((int)$pageLimit);
+
```

Comment:
```suggestion
```
i would suggest this to be removed and changing

 https://github.com/pimcore/ecommerce-framework-bundle/blob/e6ca0e41fc37f22ffafec47660ea3faca975d66b/src/FilterService/ListHelper.php#L55-L61

 to something like

`$pageLimit = (int) ($params['perPage'] ?? $filterDefinition->getPageLimit() ?? 50);`

then it could be safely considered as `int` without needing to cast to `int` again


Same for `limitOnFirstLoad` and remove the casts/intval and there
https://github.com/pimcore/ecommerce-framework-bundle/blob/e6ca0e41fc37f22ffafec47660ea3faca975d66b/src/FilterService/ListHelper.php#L83
https://github.com/pimcore/ecommerce-framework-bundle/blob/e6ca0e41fc37f22ffafec47660ea3faca975d66b/src/FilterService/ListHelper.php#L80